src/Resources/ResourceManager/Implementation/CmdletBase/DeploymentStacksCreateCmdletBase.cs

```diff
@@ -138,56 +252,129 @@ protected DeploymentStacksCreateCmdletBase()
+                }
+                else
+                {
+                    // To not break previous behavior, just output a warning.
+                    WriteWarning("${templateParameterFilePath} does not exist");
+                }
             }
-            else
+
+            // Load in dynamic parameters that were provided. They will override 
+            // parameters passed in other ways:
+            var dynamicParams = GetUsedDynamicParametersAsDictionary();
+            foreach (var param in dynamicParams)
             {
-                TemplateUri = filePath;
+                parameterObject[param.Key] = new Hashtable { { "value", param.Value } };
             }
+
+            return parameterObject;
         }
 
-        protected Hashtable ResolveParameters()
+        /// <summary>
+        /// Attempts to build the bicep param file. If no other template file, uri, or spec is present, a template or template spec
+        /// will try to be extracted from the bicep param build output.
+        /// </summary>
+        /// <param name="emitWarnings">Choice for whether to emit bicep build warnings.</param>
+        /// <exception cref="PSInvalidOperationException"></exception>
+        protected void BuildAndUseBicepParameters(bool emitWarnings)
         {
-            var output = GetParametersAndMetadataFromFile();
+            BicepUtility.OutputCallback nullCallback = null;
+            // Whatever currently used dynamic parameters are set will be used as override parameters when building the parameter file:
+            var output = BicepUtility.Create().BuildBicepParamFile(this.ResolvePath(TemplateParameterFile), GetUsedDynamicParametersAsDictionary(), this.WriteVerbose, emitWarnings ? this.WriteWarning : nullCallback);
+            bicepparamFileParameters = TemplateUtility.ParseTemplateParameterJson(output.parametersJson);
 
-            if (string.IsNullOrEmpty(TemplateFile) && 
-                string.IsNullOrEmpty(TemplateUri) && 
+            // If there is no template provided, extraction from the bicep parameter output should be attempted:
+            if (TemplateObject == null &&
+                string.IsNullOrEmpty(TemplateFile) &&
+                string.IsNullOrEmpty(TemplateUri) &&
                 string.IsNullOrEmpty(TemplateSpecId))
             {
                 // When .bicepparam support was first introduced, we were missing the validation to block overriding the 'using' path in these cmdlets.
                 // We need to be careful to retain this behavior to avoid breaking existing users, until it is re-introduced intentionally with https://github.com/Azure/bicep/issues/10333.
 
                 if (!string.IsNullOrEmpty(output.templateJson))
                 {
-                    TemplateJson = output.templateJson;
+                    TemplateObject = JsonConvert.DeserializeObject<Hashtable>(output.templateJson);
```

Comment:
@anthony-c-martin Yesterday I merged the PR [#24282] and you replaced JsonConvert.DeserializeObject<Hashtable>(output.templateJson) to output.templateJson.FromJson<Hashtable>(). Could you please confirm if it should apply to this line of code?

Reply:
Yes they'll need to be integrated here - I'm guessing this was missed in a rebase. @dantedallag please could you incorporate them?